openlibrary/components/MergeUI/utils.js

```diff
@@ -230,6 +230,38 @@ function save_many(items, comment, action, data) {
     });
 }
 
+/**
+ * Fetches name associated with the author key
+ * @param {Object[]} works
+ * @returns {Promise<Record<string,object>} A response to the request
+ */
+export async function get_author_names(works) {
+    const authorIds = _.uniq(works).flatMap(record =>
+        record.authors.map(authorEntry => authorEntry.author.key)
+    )
+
+    const queryParams = new URLSearchParams({
+        q: `key:(${authorIds.join(' OR ')})`,
+        mode: 'everything',
+        fields: 'key,name',
+    })
+
+    const response = await fetch(`${CONFIGS.OL_BASE_BOOKS}/search/authors.json?${queryParams}`)
+
+    if (!response.ok) return {error: true};
```

Comment:
Let's have this `throw` a new error, and then in the place we call it, wrap the call to this method in a try catch that...